pkg/cover/backend/dwarf.go

```diff
@@ -351,7 +351,36 @@ type pcFixFn = (func([2]uint64) ([2]uint64, bool))
 func readTextRanges(debugInfo *dwarf.Data, module *vminfo.KernelModule, pcFix pcFixFn) (
 	[]pcRange, []*CompileUnit, error) {
 	var ranges []pcRange
-	var units []*CompileUnit
+	unitMap := map[string]*CompileUnit{}
+	addRange := func(r [2]uint64, unitName string) {
+		if pcFix != nil {
+			var filtered bool
+			r, filtered = pcFix(r)
+			if filtered {
+				return
+			}
+		}
+		// Rust compilation units have a special naming scheme.
+		if strings.Contains(unitName, "/@/") {
+			unitName, _, _ = strings.Cut(unitName, "/@/")
+		}
+		unit, ok := unitMap[unitName]
```

Comment:
Is it the case that several dwarf.TagCompileUnit can produce the same unitName for Rust code due to inlininig?

Reply:
Yes, kind of. We have several Rust compile units anyway, e.g.
* `root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/lib.rs/@/core.64aa0a46a0f8f964-cgu.0`
* `rust/compiler_builtins.rs/@/compiler_builtins.77be2ccce25203a-cgu.0`
* `rust/kernel/lib.rs/@/kernel.2bb770ae1dba3d33-cgu.0`
* etc

Their line coverage references `/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/*` files, sometimes the same ones.

And even kernel code (from the `rust/*` folder) is sometimes included into multiple compilation units.